demo/src/components/canvas3d.js

```diff
@@ -0,0 +1,211 @@
+/* global window */
+import React, {Component, PropTypes} from 'react';
+import autobind from 'autobind-decorator';
+import {PerspectiveViewport} from 'deck.gl';
+
+/* Utils */
+
+// constrain number between bounds
+function clamp(x, min, max) {
+  if (x < min) {
+    return min;
+  }
+  if (x > max) {
+    return max;
+  }
+  return x;
+}
+
+// vector operations
+function add(v0, v1) {
+  return [
+    v0[0] + v1[0],
+    v0[1] + v1[1],
+    v0[2] + v1[2]
+  ];
+}
+
+function minus(v0, v1) {
+  return [
+    v0[0] - v1[0],
+    v0[1] - v1[1],
+    v0[2] - v1[2]
+  ];
+}
+
+// rotate vector around x axis
+// @param {number} angle: in degrees
+function rotateX(vector, angle) {
+  const r = angle / 180 * Math.PI;
+  const cosA = Math.cos(r);
+  const sinA = Math.sin(r);
+  const [x, y, z] = vector;
+  return [
+    x,
+    y * cosA - z * sinA,
+    y * sinA + z * cosA
+  ];
+}
+
+// rotate vector around y axis
+// @param {number} angle: in degrees
+function rotateY(vector, angle) {
+  const r = angle / 180 * Math.PI;
+  const cosA = Math.cos(r);
+  const sinA = Math.sin(r);
+  const [x, y, z] = vector;
+  return [
+    x * cosA + z * sinA,
+    y,
+    -x * sinA + z * cosA
+  ];
+}
+
+const ua = typeof window.navigator !== 'undefined' ?
+  window.navigator.userAgent.toLowerCase() : '';
+const firefox = ua.indexOf('firefox') !== -1;
+
+/* Interaction */
+
+export default class Canvas3D extends Component {
```

Comment:
> This should be named as FreeSpaceViewportController or something similar

Is this an "orbit controller" as in THREE.js etc? I.e. we can move around in an orbit always looking towards the center?